openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,464 @@
+            get_template_data_from_session)
+        anatomy_data = get_template_data_from_session()
+
+    # add project roots to anatomy data
+    anatomy_data["root"] = anatomy.roots
+    anatomy_data["platform"] = platform.system().lower()
+
+    # get colorspace settings
+    imageio_global, imageio_host = _get_imageio_settings(
+        project_settings, host_name)
+
+    config_host = imageio_host["ocio_config"]
+
+    if config_host["enabled"]:
+        print(config_host["filepath"])
+        config_data = _get_config_data(
+            config_host["filepath"], anatomy_data
+        )
+    else:
+        config_data = None
+
+    if not config_data:
+        # get config path from either global or host_name
+        config_global = imageio_global["ocio_config"]
+        print(config_global["filepath"])
+        config_data = _get_config_data(
+            config_global["filepath"], anatomy_data
+        )
+
+    if not config_data:
+        raise FileExistsError(
+            "No OCIO config found in settings. It is "
+            "either missing or there is typo in path inputs"
+        )
+
+    return config_data
+
+
+def _get_config_data(path_list, anatomy_data):
+    """Path formating in interation
+
+    Args:
+        path_list (list[str]): list of abs paths
+        anatomy_data (dict): formating data
+
+    Returns:
+        dict: config data
+    """
+    # first try host config paths
+    for path_ in path_list:
+        formated_path = _format_path(path_, anatomy_data)
+        if not os.path.exists(formated_path):
+            continue
+
+        return {
+            "path": os.path.normpath(formated_path),
+            "template": path_
+        }
+
+
+def _format_path(tempate_path, anatomy_data):
+    """Single template path formating.
+
+    Args:
+        tempate_path (str): template string
+        anatomy_data (dict): formating data
+
+    Returns:
+        str: absolute formated path
+    """
+    formatting_data = deepcopy(anatomy_data)
+
+    # format the path for potential env vars
+    formatting_data.update(dict(**os.environ))
+
+    # format path for anatomy keys
+    formatted_path = StringTemplate(tempate_path).format(
+        formatting_data)
+
+    return os.path.abspath(formatted_path)
+
+
+def get_imageio_file_rules(project_name, host_name, project_settings=None):
+    """Get ImageIO File rules from project settings
+
+    Args:
+        project_name (str): project name
+        host_name (str): host name
+        project_settings (dict, optional): project settings.
+                                           Defaults to None.
+
+    Returns:
+        dict: file rules data
+    """
+    project_settings = project_settings or get_project_settings(project_name)
+
+    imageio_global, imageio_host = _get_imageio_settings(
+        project_settings, host_name)
+
+    # get file rules from global and host_name
+    frules_global = imageio_global["file_rules"]
+    frules_host = imageio_host["file_rules"]
```

Comment:
As far as I know there is no hard requirement for any host - even if rudimentary to have implemented `imageio` in settings. Right? This might need to be me safeguarded to not exist - see comment below.

Reply:
The idea here was to make all DCC integrations with pixel data support required to have imageio shemas in settings. Weather they use them is other thing. But for example, even Photoshop would have available settings in config and file rules. Then it can use the config for some of its own config and file rules matching the config or file rules would be referencing the globally defined config. So I would rather keep this as required.

Reply:
Unless you implement those settings in this PR, I think you should not have it as required but instead warn about missing settings.